src/coreclr/inc/nsutilpriv.h

```diff
@@ -224,20 +150,15 @@ void MakeNestedTypeName(                // throws on out of memory
     ns::MakePath(toptr, __i##toptr, pnamespace, pname); \
 }
 
-#define MAKE_FULL_PATH_ON_STACK_UNICODE(toptr, pnamespace, pname) \
-{ \
-    int __i##toptr = ns::GetFullLength(pnamespace, pname); \
-    toptr = (WCHAR *) alloca(__i##toptr * sizeof(WCHAR)); \
-    ns::MakePath(toptr, __i##toptr, pnamespace, pname); \
-}
-
-#define MAKE_FULLY_QUALIFIED_NAME(pszFullyQualifiedName, pszNameSpace, pszName) MAKE_FULL_PATH_ON_STACK_UTF8(pszFullyQualifiedName, pszNameSpace, pszName)
-
+#ifdef _PREFAST_
+// need to set ptr=NULL so we don't get a build error because ptr isn't inited in a couple cases
+#define MAKE_FULLY_QUALIFIED_MEMBER_NAME(ptr, pszNameSpace, pszClassName, pszMemberName, pszSig) ptr=NULL;
+#else //_PREFAST_
 #define MAKE_FULLY_QUALIFIED_MEMBER_NAME(ptr, pszNameSpace, pszClassName, pszMemberName, pszSig) \
 { \
     int __i##ptr = ns::GetFullLength(pszNameSpace, pszClassName); \
     __i##ptr += (pszMemberName ? (int) strlen(pszMemberName) : 0); \
-    __i##ptr += (NAMESPACE_SEPARATOR_LEN * 2); \
+    __i##ptr += (int)strlen(NAMESPACE_SEPARATOR_STR) + 1; \
```

Comment:
This change does not look right. The code below adds `NAMESPACE_SEPARATOR_STR` up to two times, so it makes sense that `NAMESPACE_SEPARATOR_LEN` was multiplied by two.